### PR TITLE
Update usage and schema documentation

### PR DESCRIPTION
**Pull request by kylebarron**
### Change list

- Allow `pystac.Item` input wherever `dict` is allowed
- Add doc page on schema considerations
- Add usage doc page with high-level considerations for converting between JSON, dicts, Arrow, Parquet, Delta Lake

### CODE REVIEW

**Comment by kylebarron** [comment]
I published the docs (triggered docs CI manually) from this branch so we can view the rendered docs. We can delete the 0.5.1 version from the docs when we publish the 0.6 release version

File: docs/usage.md [hunk 1]
```diff
@@ -1 +1,50 @@
 # Usage
+
+Except for the [legacy API](api/legacy.md), [Apache Arrow](https://arrow.apache.org/) is used as the in-memory interchange format between all formats. While some end-to-end helper functions are provided, the user can go through Arrow objects for maximal flexibility in the conversion process.
```
Comment (TomAugspurger) [inline docs/usage.md:3]: Not even worth mentioning that legacy API, I think.

```suggestion
[Apache Arrow](https://arrow.apache.org/) is used as the in-memory interchange format between all formats. While some end-to-end helper functions are provided, the user can go through Arrow objects for maximal flexibility in the conversion process.
```

File: docs/usage.md [hunk 2]
```diff
@@ -1 +1,50 @@
...
+
+All functionality that goes through Arrow is currently exported via the `stac_geoparquet.arrow` namespace.
```
Comment (TomAugspurger) [inline docs/usage.md:5]: I think that an example here (or on the index page?) from STAC -> geoparquet would be valuable.

We have the one currently at https://github.com/stac-utils/stac-geoparquet?tab=readme-ov-file#usage. Even an example using a simple `pystac.Item()` built locally would be good to have.

File: docs/usage.md [hunk 3]
```diff
@@ -1 +1,50 @@
...
+
+[`parse_stac_ndjson_to_arrow`][stac_geoparquet.arrow.parse_stac_ndjson_to_arrow] is a helper function to take one or more JSON or newline-delimited JSON files on disk, infer the schema from all of them, and convert the data to a stream of Arrow record batches.
+
+### Convert Arrow to `dict`s
+
+Use [`stac_table_to_items`][stac_geoparquet.arrow.stac_table_to_items] to convert a table or stream of Arrow record batches of STAC data to a generator of Python `dict`s. This accepts either a `pyarrow.Table` or a `pyarrow.RecordBatchReader`, which allows conversions of larger-than-memory files in a streaming manner.
+
+### Convert Arrow to JSON
+
+Use [`stac_table_to_ndjson`][stac_geoparquet.arrow.stac_table_to_ndjson] to convert a table or stream of Arrow record batches of STAC data to a generator of Python `dict`s. This accepts either a `pyarrow.Table` or a `pyarrow.RecordBatchReader`, which allows conversions of larger-than-memory files in a streaming manner.
```
Comment (TomAugspurger) [inline docs/usage.md:23]: > to a generator of Python `dict`s.

Copy-paste error maybe? IIUC, this goes to an output ndjson file?

File: stac_geoparquet/arrow/_batch.py [hunk 4]
```diff
@@ -83,6 +87,9 @@ def from_dicts(
         # `ArrowInvalid: cannot mix list and non-list, non-null values`
         wkb_items = []
         for item in items:
+            if isinstance(item, pystac.Item):
```
Comment (TomAugspurger) [inline stac_geoparquet/arrow/_batch.py:90]: At some point we'll want to profile this. I'm guessing that the item copying and shapely stuff is going to be slower than `isinstance`, but it's a little unfortunate to pay the cost. We could avoid that with a `from_pystac_items` which takes a an `Iterable[pystac.Item]`.

**Review by TomAugspurger — APPROVED** [review]

File: docs/usage.md [hunk 5]
```diff
@@ -1 +1,50 @@
...
+
+## Parquet
+
+Use [`to_parquet`][stac_geoparquet.arrow.to_parquet] to write STAC Arrow data in memory. This is a special function to ensure that [GeoParquet](https://geoparquet.org/) 1.0 or 1.1 metadata is written to the Parquet file.
```
Comment (gadomski) [inline docs/usage.md:27]:
```suggestion
Use [`to_parquet`][stac_geoparquet.arrow.to_parquet] to write STAC Arrow data from memory to a path or file-like object. This is a special function to ensure that [GeoParquet](https://geoparquet.org/) 1.0 or 1.1 metadata is written to the Parquet file.
```

File: stac_geoparquet/arrow/_batch.py [hunk 6]
```diff
@@ -83,6 +87,9 @@ def from_dicts(
...
+                item = item.to_dict()
```
Comment (gadomski) [inline stac_geoparquet/arrow/_batch.py:91]: Re: https://github.com/stac-utils/pystac/issues/960, you probably want to ensure that you don't try to resolve the root link (this is an annoying **pystac** quirk):

```suggestion
                item = item.to_dict(tranform_hrefs=False)
```

File: stac_geoparquet/arrow/_batch.py (hunk 4 above)
Comment (kylebarron) [reply stac_geoparquet/arrow/_batch.py:90]: I can remove the pystac functionality from this PR if you prefer

Comment (TomAugspurger) [reply stac_geoparquet/arrow/_batch.py:90]: Took a quick look, and I think this is cheap enough relative to the other stuff to be fine:

```python
In [8]: import shapely, pystac

In [9]: item = pystac.read_file("https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20230112T104411_R008_T29NPE_20230113T053333")

In [11]: %timeit isinstance(item, pystac.Item)
64.2 ns ± 8.36 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [12]: %timeit shapely.to_wkb(shapely.geometry.shape(item.geometry), flavor="iso")
36.7 µs ± 4.78 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

File: docs/usage.md (hunk 2 above)
Comment (kylebarron) [reply docs/usage.md:5]: I added a simple example below

**Pull request merged**